File: ITPMApp（第１回 実演時作成）/app/src/main/java/com/example/itpm_app/databases/ITPMDataOpenHelper.java
package com.example.itpm_app.databases;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;

public class ITPMDataOpenHelper extends SQLiteOpenHelper {

    private static final String DATABASE_NAME = "itpmDB.db";
    private static final int DATABASE_VERSION = 1;

    public static final String TABLE_NAME = "itpmdb";
    public static final String COLUMN_ID = "_id";
    public static final String COLUMN_TITLE = "title";

    private static final String CREATE_TABLE ="CREATE TABLE " + TABLE_NAME + " (" + COLUMN_ID + " INTEGER PRIMARY KEY, " + COLUMN_TITLE + " TEXT)";
    private static final String INIT_TABLE = "INSERT INTO " + TABLE_NAME + " VALUES (1,'ホーム'),(2,'事業内容'),(3,'企業情報'),(4,'採用情報'),(5,'お問い合わせ')";
    private static final String DROP_TABLE = "DROP TABLE IF EXISTS " + TABLE_NAME;

    public ITPMDataOpenHelper(@Nullable Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        // テーブル作成
        db.execSQL(CREATE_TABLE);
        // 初期データのインサート
        db.execSQL(INIT_TABLE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int i, int i1) {
        // テーブル削除
        db.execSQL(DROP_TABLE);
        onCreate(db);
    }
}

File: ITPMApp（第１回 実演時作成）/kanpelibrary/src/main/java/com/example/kanpelibrary/databases/ITPMDataOpenHelper.kt
package com.example.kanpelibrary.databases

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class ITPMDataOpenHelper(context: Context): SQLiteOpenHelper(context,
    DATABASE_NAME, null,
    DATABASE_VERSION
) {

    companion object {
        private const val DATABASE_NAME = "itpmDB.db"
        private const val DATABASE_VERSION = 1
        const val TABLE_NAME = "itpmdb"
        const val _ID = "_id"
        const val COLUMN_TITLE = "title"

        private const val CREATE_TABLE = "CREATE TABLE $TABLE_NAME ($_ID INTEGER PRIMARY KEY, $COLUMN_TITLE TEXT)"
        private const val INIT_TABLE = "INSERT INTO $TABLE_NAME VALUES (1,'ホーム'),(2,'事業内容'),(3,'企業情報'),(4,'採用情報'),(5,'お問い合わせ')"
        private const val DROP_TABLE = "DROP TABLE IF EXISTS $TABLE_NAME"
    }

    override fun onCreate(db: SQLiteDatabase?) {
        db?.execSQL(CREATE_TABLE)
        db?.execSQL(INIT_TABLE)
    }

    override fun onUpgrade(db: SQLiteDatabase?, p1: Int, p2: Int) {
        db?.execSQL(DROP_TABLE)
        onCreate(db)
    }
}